.github/workflows/deploy.yml

```
@@ -0,0 +1,279 @@
+      contents: write
+      packages: write
+      
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          
+      - name: Determine deployment parameters
+        id: params
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2129:style:17:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)